File: backend/Services/Parsers/TravleParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class TravleParser : BasicScoreResultParser
    {
        private ILogger<TravleParser> _logger;

        public TravleParser(ILogger<TravleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string? HelpText => null;
        public override string GameName => "Travle";
        protected override Regex Parser => new Regex($@"#{GameName.ToLower()} #\d+ \+?(?<{ScoreGroup}>\d*)");
        protected override string? ExtraContent => Url;
        public override string Url => "https://travle.earth";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults) {
            var newResult = base.SetScore(dailyResult, parserResults);

            if (dailyResult.Result != null && dailyResult.Result.Contains("(Perfect)")) {
                newResult.Score = -1;
            }

            return newResult;
        }
    }
}

File: lambdas/src/parsers/travle-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';

export class TravleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Travle';
  readonly helpText = null;
  readonly url = 'https://travle.earth';
  protected readonly extraContent = this.url;
  
  // Matches "#travle #742 (1 away)" or "#travle #555 +0 (Perfect)" or "#travle #535 +6"
  // The regex captures optional +X score, may be empty for failures like "(1 away)"
  protected readonly parser = new RegExp(
    `#${this.gameName.toLowerCase()} #\\d+ \\+?(?<${BasicScoreResultParser.SCORE_GROUP}>\\d*)`
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    // First call base to set score from +X
    const newResult = super.setScore(dailyResult, match);
    
    // Check for (Perfect) in result to set score to -1
    if (dailyResult.result && dailyResult.result.includes('(Perfect)')) {
      newResult.score = -1;
    }
    
    return newResult;
  }
}
